watchers/target_pod_test.go

```diff
@@ -0,0 +1,77 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023 Datadog, Inc.
+
+package watchers_test
+
+import (
+	chaosv1beta1 "github.com/DataDog/chaos-controller/api/v1beta1"
+	"github.com/DataDog/chaos-controller/log"
+	"github.com/DataDog/chaos-controller/metrics/noop"
+	"github.com/DataDog/chaos-controller/watchers"
+	"github.com/DataDog/chaos-controller/watchers/mocks"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"go.uber.org/zap"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+var _ = Describe("target pod", func() {
+	Describe("Handler", func() {
+		//	@ToDo
+	})
+
+	Describe("NewDisruptionTargetWatcher", func() {
+		const watcherName = "name"
+
+		var (
+			disruption chaosv1beta1.Disruption
+			err        error
+			watcher    watchers.Watcher
+		)
+
+		JustBeforeEach(func() {
+			// Arrange
+			var logger *zap.SugaredLogger
+			logger, err = log.NewZapLogger()
+			Expect(err).ShouldNot(HaveOccurred())
+
+			// Action
+			watcher, err = watchers.NewDisruptionTargetWatcher(watcherName, &mocks.EventRecorderMock{}, &mocks.ReaderMock{}, true, &disruption, &noop.Sink{}, logger)
+		})
+
+		Context("with a valid disruption", func() {
+			BeforeEach(func() {
+				// Arrange
+				disruption = chaosv1beta1.Disruption{
+					Spec: chaosv1beta1.DisruptionSpec{
+						Selector: labels.Set{
+							"Lorem": "ipsum",
+						},
+					},
+				}
+			})
+
+			It("should not return an error", func() {
+				Expect(err).ShouldNot(HaveOccurred())
+			})
```

Comment:
FYI should succeed also exists (and does the same)
```suggestion
			It("should succeed", func() {
				Expect(err).Should(Succeed())
			})
```

Reply:
For me, `Expect(err).Should(Succeed())` means the error has been raised. I prefer `ShouldNot(HaveOccurred())`, for my it is less ambiguous.

Reply:
Yeah, with the error here it's definitely misleading, this helper has been introduced when you have the method call into the expectation:

```golang
Expect(MyMethodReturnOnlyErrorOrNil()).Should(Succeed())
```

Here it's readable, with only `err` it's definitely not.

Thanks for challenging 🤗

Reply:
I agree with `Expect(MyMethodReturnOnlyErrorOrNil()).Should(Succeed())` because the action should succeed in a nominal case. Thanks for your comprehension :bow: